exec/java-exec/src/main/java/org/apache/drill/exec/physical/config/HashJoinPOP.java

```diff
@@ -32,35 +33,53 @@
 import com.google.common.base.Preconditions;
 import com.google.common.collect.Lists;
 import org.apache.drill.exec.physical.base.AbstractJoinPop;
+import org.apache.drill.exec.work.filter.RuntimeFilterDef;
+
 
 @JsonTypeName("hash-join")
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class HashJoinPOP extends AbstractJoinPop {
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(HashJoinPOP.class);
 
+  private RuntimeFilterDef runtimeFilterDef;
```

Comment:
Who is responsible for constructing this runtimeFilterDef?

Reply:
After physical plan is generated, RuntimeFilterManager is is responsible for constructing this runtimeFilterDef. So I guess your estimation of the cost sending bloom filter is trivial.

Reply:
RuntimeFilterDef was created after the physical plan was generated. Constructing the BloomFilter at HashJoin node is a heavy work， to a light weight join query (i.e. two small tables), it's not applicable to construct a bloom filter.So I left some TODO work at the RuntimeFilterManager.quelifiedHolders method to value whether applicable to apply the RuntimeFilter according to the NDV from both sides. The NDV statistics is not supported now. I will fire another work about the Metadata to support it.